### PR TITLE
[macos] Fix msbuild issues with shprojj/projitem library

### PR DESCRIPTION
**Pull request by chamons**
- In some build cases this chunk of code:

    <ItemGroup Condition=" '$(NoCompilerStandardLib)' == 'true' and '$(NoStdLib)' != 'true' ">
          
        <_ExplicitReference Include="$(FrameworkPathOverride)\mscorlib.dll" />
    </ItemGroup>

would trigger and force us to use mscorlib from system mono. That does not work well.
- By setting FrameworkPathOverride, we can get the right mscorlib
- However, that ItemGroup happens outside of a target, so we must move our setting to match for it to take effect

### CODE REVIEW

**Comment by chamons** [comment]
Added VS guys since it is an msbuild change.

I forgot to add you to these two msbuild changes that happened in recent past:

https://github.com/xamarin/xamarin-macios/pull/1846
https://github.com/xamarin/xamarin-macios/pull/1827

apologies.

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.msbuild.targets [hunk 1]
```diff
@@ -16,11 +16,12 @@ Copyright (c) 2017 Microsoft Corp. (www.microsoft.com)
 
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 	<!-- MSBuild specific hacks - Teach msbuild where to find our BCL and facades -->
+	<PropertyGroup>
```
Comment (kzu) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.msbuild.targets]: This should have a `Condition="'$(OS)' != 'Win_NT'"`

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.msbuild.targets [hunk 2]
```diff
@@ -16,11 +16,12 @@ Copyright (c) 2017 Microsoft Corp. (www.microsoft.com)
...
+		<MacBclPath Condition="'$(TargetFrameworkIdentifier)' != 'Xamarin.Mac'">$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
+		<MacBclPath Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.Mac'">$(XamarinMacFrameworkRoot)/lib/mono/Xamarin.Mac</MacBclPath>
+		<FrameworkPathOverride>$(MacBclPath)</FrameworkPathOverride>
+	</PropertyGroup>
 	<Target Name="FixTargetFrameworkDirectory" AfterTargets="GetReferenceAssemblyPaths">
```
Comment (kzu) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.msbuild.targets]: Ditto here: `Condition="'$(OS)' != 'Win_NT'"`

**Comment by chamons** [comment]
@kzu Changes made.

**Comment by chamons** [comment]
Known random failure - https://bugzilla.xamarin.com/show_bug.cgi?id=52968

**Pull request merged**